cin_validator/rules/cin2022_23/rule_8867.py

```diff
@@ -0,0 +1,243 @@
+
+CINdetails = CINTable.CINdetails
+CINclosureDate = CINdetails.CINclosureDate
+LAchildID = CINdetails.LAchildID
+CINdetailsID = CINdetails.CINdetailsID
+
+Assessments = CINTable.Assessments
+AssessmentAuthorisationDate = Assessments.AssessmentAuthorisationDate
+LAchildID = Assessments.LAchildID
+CINdetailsAssID = Assessments.CINdetailsID
+
+# define characteristics of rule
+@rule_definition(
+    code=8867,
+    module=CINTable.CINdetails,
+    message="CIN episode is shown as closed, however Assessment is not shown as completed",
+    affected_fields=[
+        CINclosureDate,
+        AssessmentAuthorisationDate,
+    ],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # PREPARING DATA
+
+    # Replace ChildProtectionPlans with the name of the table you need.
+    df_cind = data_container[CINdetails].copy()
+    df_ass = data_container[Assessments].copy()
+
+    # Before you begin, rename the index so that the initial row positions can be kept intact.
+    df_cind.index.name = "ROW_ID"
+    df_ass.index.name = "ROW_ID"
+
+    # Resetting the index causes the ROW_IDs to become columns of their respective DataFrames
+    # so that they can come along when the merge is done.
+    df_cind.reset_index(inplace=True)
+    df_ass.reset_index(inplace=True)
+
+    # lOGIC
+    # Implement rule logic as described by the Github issue.
+    # Put the description as a comment above the implementation as shown.
+
+    #If <CINclosureDate> (N00102) is present then all instances of the <Assessments> group must include the <AssesssmentAuthorisationDate> (N00160)
+    # Issues dfs should return rows where an AssessmentAuthorisationDate exists if a CINclosureDate has been recorded for that child.
+
+    #  Create dataframes which only have rows with a CINclosureDate and an AssessmentAuthorisation Date and which should have one plan per row.
+    df_cind = df_cind[df_cind[CINclosureDate].notna()]
+    
+    #  Merge tables to get corresponding CP plan group and reviews
+    df_merged = df_cind.merge(
+        df_ass,
+        left_on=["LAchildID", "CINdetailsID"],
+        right_on=["LAchildID", "CINdetailsID"],
+        how="left",
+        suffixes=("_cind", "_ass"),
+    )
+
+    #  Get rows where there is no AssessmentAuthorisationDate when a CINclosureDate is recorded. 
+    condition = df_merged[AssessmentAuthorisationDate].isna()
+    df_merged = df_merged[condition].reset_index()
+
+    # create an identifier for each error instance.
+    # In this case, the rule is checked for each CINclosureDate and AssessmentAuthorisationDate in each child (differentiated by LAchildID)
+    # So, a combination of LAchildID, CINclosureDate and AssessmentAuthorisationDate identifies and error instance.
+    df_merged["ERROR_ID"] = tuple(
+        zip(df_merged[LAchildID], df_merged[CINdetailsID], df_merged[CINclosureDate], df_merged[AssessmentAuthorisationDate])
+    )
+
```

Comment:
Since all the failing positions will have the same value of AssessmentAuthorisationDate (that is pd.NA in this case), it doesn't add to the uniqueness of the ERROR_IDs so we can remove it. 

I'll fix the formatting, remove that column and merge it in. 

Thank you @GBurrel,
This is good work.